File: finance-app-backend/src/main/java/com/victor/financeapp/backend/application/dto/InvoiceDTO.java
package com.victor.financeapp.backend.application.dto;

import java.time.YearMonth;
import java.util.List;

public record InvoiceDTO(
        Long id,
        Long creditCardId,
        Boolean isClosed,
        YearMonth yearMonth,
        List<TransactionDTO> transactions
) {
}

File: finance-app-android/app/src/main/java/com/viictrp/financeapp/application/dto/InvoiceDTO.kt
package com.viictrp.financeapp.application.dto

import java.time.YearMonth

data class InvoiceDTO(
    val id: Long,
    val creditCardId: Long,
    val transactions: List<TransactionDTO> = emptyList(),
    val isClosed: Boolean,
    val yearMonth: YearMonth
)
